pkg/windows/service/service.go

```diff
@@ -0,0 +1,235 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	"github.com/golang/glog"
+
+	"github.com/spf13/pflag"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+var (
+	flServiceName        string
+	flServiceDisplayName string
+	flRegisterService    bool
+	flUnregisterService  bool
+	flRunService         bool
+
+	service *handler
+)
+
+type handler struct {
+	tosvc   chan bool
+	fromsvc chan error
+}
+
+func getServicePath() (string, error) {
+	p, err := exec.LookPath(os.Args[0])
```

Comment:
Does `os.Args[0]` here work if the kubelet is started as a hyperkube subcommand?
(does hyperkube support windows?)

Reply:
To be honest I have no idea if hyperkube is supported on Windows. I don't understand who will try to register the service via hyperkube. It kind of defeats the point of both.

Reply:
I'm not aware of any hyperkube support for Windows, and running the kubelet in a container won't work because `--privileged` isn't available on Windows. It needs to be run on the host directly.

Reply:
@deads2k do you know?

Reply:
> @deads2k do you know?

I do not know.